sdk/formrecognizer/ai-form-recognizer/review/ai-form-recognizer.api.md

```diff
@@ -0,0 +1,1299 @@
+                                readonly type: "number";
+                                readonly description: "Tax associated with each line item. Possible values include tax amount, tax %, and tax Y/N";
+                                readonly example: "10%";
+                            };
+                            readonly Unit: {
+                                readonly type: "string";
+                                readonly description: "The unit of the line item, e.g, kg, lb etc.";
+                                readonly example: "hours";
+                            };
+                            readonly UnitPrice: {
+                                readonly type: "number";
+                                readonly description: "The net or gross price (depending on the gross invoice setting of the invoice) of one unit of this item";
+                                readonly example: "$30.00";
+                            };
+                        };
+                    };
+                };
+            };
+        };
+    };
+};
+
+// @public
+export interface LayoutResult {
+    // (undocumented)
+    pages: DocumentPage[];
+    styles: DocumentStyle[];
+    tables: DocumentTable[];
+}
+
+// @public
+export type LengthUnit = string;
+
+// @public
+export interface ListModelsOptions extends OperationOptions {
+}
+
+// @public
+export interface ListOperationsOptions extends OperationOptions {
+}
+
+// @public
+export type ModelInfo = ModelSummary & {
+    docTypes?: {
+        [propertyName: string]: DocTypeInfo;
+    };
+};
+
+// @public
+export interface ModelSchema {
+    docTypes: {
+        [type: string]: {
+            fieldSchema: {
+                [k: string]: FieldSchema;
+            };
+        };
+    };
+    modelId: string;
+}
+
+// @public
+export interface ModelSummary {
+    createdDateTime: Date;
+    description?: string;
+    modelId: string;
+}
+
+// @public
+export interface NumberFieldSchema<Type extends "number" | "integer" = "number" | "integer"> {
+    // (undocumented)
+    readonly type: Type;
+}
+
+// @public
+export interface ObjectFieldSchema<Properties extends {
+    readonly [k: string]: FieldSchema;
+} = {
+    readonly [k: string]: FieldSchema;
+}> {
+    readonly properties: Properties;
+    // (undocumented)
+    readonly type: "object";
+}
+
+// @public
+export interface OperationInfo {
+    createdDateTime: Date;
+    kind?: OperationKind;
+    lastUpdatedDateTime: Date;
+    operationId: string;
+    percentCompleted?: number;
+    resourceLocation: string;
+    status: OperationStatus;
+}
+
+// @public
+export type OperationKind = string;
+
+// @public
+export type OperationStatus = "notStarted" | "running" | "failed" | "succeeded" | "canceled";
+
+// @public
+export interface PollerOptions<TState extends PollOperationState<unknown>> {
+    onProgress?: (state: TState) => void;
+    resumeFrom?: string;
+    updateIntervalInMs?: number;
+}
```

Comment:
Why does the poller options extend the state?

Reply:
The generic type `TState` is constrained to be a PollOperationState, but the options type does not extend it.

Reply:
I can't read 2x 😆

Another question, does not the poller (`PollerLike`) already has an onProgress handler? why do you duplicate one here?

Reply:
This is required to handle the progress of the first state update, which happens within the call to the method that creates the poller. If a customer can only use `onProgress` on the poller itself, then they cannot capture the first state update, so in previous FR iterations we also had `onProgress` as a member of the polling options.

Reply:
The only thing really to clarify is that this is an initial `onProgress` handler. It'll be added before the poller is returned. I can add that to the docstring.